basic-steps/src/main/resources/org/jenkinsci/plugins/workflow/steps/CatchErrorStep/help.html

```diff
@@ -1,32 +1,31 @@
 <div>
-    <p>
     If the body throws an exception, mark the build as a failure, but nonetheless
-    continue to execute the workflow from the statement next to the 'catchError' step.
-    This is useful if you want to find as many errors as you can before terminating
-    the workflow. See the following example:
-
-<pre>
-catchError {
-    sh 'false'  // this step will fail
-    
-    echo 'this step will not run'
+    continue to execute the workflow from the statement following the <code>catchError</code> step.
+    This is <em>only</em> necessary when using certain post-build actions (notifiers)
+    originally defined for freestyle projects which pay attention to the result of the ongoing build.
+<p><pre>
+node {
+    catchError {
+        sh 'might fail'
+    }
+    step([$class: 'Mailer', recipients: 'admin@somewhere'])
 }
-echo 'but this step will run after sh fails'
-...
-// and the workflow fails at this point after normally completing the program
 </pre>
-    
-    <p>
-    If you want to ignore a failure, use the try/catch block instead:
-
-<pre>
-try {
-    sh 'false'  // this step will fail
-
-    echo 'this step will not run'
-} catch(_) {}
-echo 'but this step will run after sh fails'
-...
-// and the workflow as a whole would succeed
+    <p>If the shell step fails, the flow build’s status will be set to failed, so that the subsequent mail step will see that this build is failed.
+    In the case of the mail sender, this means that it will send mail.
+    (It may also send mail if this build <em>succeeded</em> but previous ones failed, and so on.)
+    Even in that case, this step can be replaced by the following idiom:
+<p><pre>
+node {
+    try {
+        sh 'might fail'
+    } catch (err) {
+        echo "Caught: ${err}"
+        currentBuild.result = 'FAILURE'
+    }
+    step([$class: 'Mailer', recipients: 'admin@somewhere'])
+}
 </pre>
-</div>
\ No newline at end of file
+    <p>For all other cases, use plain <code>try</code>-<code>catch</code>(-<code>finally</code>) blocks.
```

Comment:
This is where I meant to leave the code example. For some people, "plain try-catch-finally block" isn't obvious. In my original version I was trying to emphasize that by having the actual snippet.